pkg/deployer/helm/ensure.go

```diff
@@ -201,16 +201,17 @@ func (h *Helm) DeleteFiles(ctx context.Context) error {
 	}
 
 	// remove finalizer
-	controllerutil.RemoveFinalizer(&h.DeployItem.ObjectMeta, lsv1alpha1.LandscaperFinalizer)
+	controllerutil.RemoveFinalizer(h.DeployItem, lsv1alpha1.LandscaperFinalizer)
 	return h.kubeClient.Update(ctx, h.DeployItem)
 }
 
 // ApplyObject applies a managed resource to the target cluster.
 func (h *Helm) ApplyObject(ctx context.Context, kubeClient client.Client, obj *unstructured.Unstructured) error {
 	currOp := "ApplyObjects"
-	currObj := obj.NewEmptyInstance()
+	currObj := unstructured.Unstructured{}
```

Comment:
has `NewEmptyInstance` been removed?

Reply:
No, but it returns an object of type `runtime.Unstructured` which does not implement the `client.Object` interface that is now required by `kubeClient.Get`. I'm somewhat confused why `NewEmptyInstance` called on type `unstructured.Unstructured` - which implements the interface - does return a different type. This was the first workaround that I found - it basically does exactly the same thing as `NewEmptyInstance`, but with the `unstructured.Unstructured` type.

Reply:
ah makes sense.
Can you add that as a comment maybe

Reply:
done